packages/react-dev-utils/checkRequiredFiles.js

```diff
@@ -7,24 +7,51 @@
 
 'use strict';
 
-var fs = require('fs');
-var path = require('path');
-var chalk = require('chalk');
+const fs = require('fs');
+const path = require('path');
+const chalk = require('chalk');
+
+// Currently, files passed without an extension are assumed to be js-like.
+// If other types of files need to be passed later without extensions the
+// checkRequiredFiles function will need to be modified.
+const fallbackExts = [
+  '.web.js',
+  '.mjs',
+  '.js',
+  '.web.jsx',
+  '.jsx',
+  '.ts',
+  '.tsx',
+];
+const withExt = (ext, file) => file.dir + path.sep + file.name + ext;
 
 function checkRequiredFiles(files) {
-  var currentFilePath;
+  let didFallbackExts = false;
   try {
-    files.forEach(filePath => {
-      currentFilePath = filePath;
-      fs.accessSync(filePath, fs.F_OK);
-    });
+    for (const filePath of files) {
+      const file = path.parse(filePath);
+
+      if (file.ext === '') {
```

Comment:
I considered making this
```javascript
if (file.ext === '' || !fs.existsSync(filePath))
```
so that files **with** extensions that aren't found would also be checked with fallbacks. But since at the moment the only file other than `index.js` being required is a `.html` file, that seemed a bad idea. If this function sees more use in the future, that concept could be revisited.